src/state/process.rs

```diff
@@ -417,6 +421,51 @@ where
         } else {
             self.transmit(OverlordMsg::SignedVote(signed_vote)).await?;
         }
+
+        // On handling the signed prevote vote, some signed votes and quorum certificates might have
+        // been cached in the vote collector. So it should check whether there is votes or quorum
+        // certificates exsits or not. If self node is not the leader, check if there is prevoteQC
+        // exits. If self node is the leader, check if there is signed prevote vote exsits. It
+        // should be noted that when self is the leader, the process after checking the vote is the
+        // same as the handle signed vote.
+        if !self.is_leader {
+            if let Ok(qc) = self
+                .votes
+                .get_qc(self.epoch_id, self.round, VoteType::Prevote)
+            {
+                self.state_machine.trigger(SMRTrigger {
+                    trigger_type: qc.vote_type.into(),
+                    source:       TriggerSource::State,
+                    hash:         qc.epoch_hash,
+                    round:        Some(self.round),
+                })?;
+                return Ok(());
+            }
+        } else if let Some(epoch_hash) = self.counting_vote(VoteType::Prevote)? {
+            let qc = self.generate_qc(epoch_hash.clone(), VoteType::Prevote)?;
+            let vote_type = VoteType::Prevote;
+            self.votes.set_qc(qc.clone());
+            self.broadcast(OverlordMsg::AggregatedVote(qc)).await?;
+
+            let set = self.full_transcation.lock();
+            let epoch_hash = if set.contains(&epoch_hash) {
+                epoch_hash
+            } else {
+                Hash::new()
```

Comment:
Is the `EpochHash` generated externally?

It is recommended to construct an empty hash concept instead of `0x00000...`, such as using `RLP ::NULL_RLP` or `[]` to build an empty hash.

Reply:
This empty `Bytes` is used for the SMR.